rules/no-array-for-each.js

```diff
@@ -314,14 +337,15 @@ function isFixable(callExpression, {scope, functionInfo, allIdentifiers, context
 	}
 
 	// Check `CallExpression.parent`
-	if (callExpression.parent.type !== 'ExpressionStatement') {
+	if (callExpression.parent.type !== 'ExpressionStatement' && callExpression.parent.type !== 'ChainExpression') {
 		return false;
 	}
 
-	// Check `CallExpression.callee`
-	// Because of `ChainExpression` wrapper, `foo?.forEach()` is already failed on previous check keep this just for safety
-	/* c8 ignore next 3 */
-	if (callExpression.callee.optional) {
+	if (isOptionalChaining && callExpression.parent.parent.type !== 'ExpressionStatement') {
+		return false;
+	}
+
+	if (isOptionalChaining && callExpression.callee.object.type !== 'Identifier' && callExpression.callee.object.type !== 'MemberExpression') {
```

Comment:
Are you serious?

`(a ? b : c)?.forEach()`
`(0, foo)?.forEach()`
`(foo || bar)?.forEach()`

Reply:
Well, that would involve `(foo && bar())?.forEach()`, `(a ? b : c()).forEach()` and other cases where one of the function calls has a side effect, hence it shouldn't be auto-fixable. AFAICS, it'd considerably complicate the parsing, so I'd say it's reasonable to keep that out of scope for this PR

Reply:
I don't think we can ignore them, it's very common.
Would you mind let me work on it?

Reply:
Sure